controllers/eventhandlers/gateway.go

```diff
@@ -119,3 +121,42 @@ func (h *enqueueRequestsForGatewayEvent) enqueueImpactedRoutes(queue workqueue.R
 		}
 	}
 }
+
+type gatewayEventHandler struct {
+	log    gwlog.Logger
+	client client.Client
+	mapper *resourceMapper
+}
+
+func NewGatewayEventHandler(log gwlog.Logger, client client.Client) *gatewayEventHandler {
+	return &gatewayEventHandler{log: log, client: client,
+		mapper: &resourceMapper{log: log, client: client}}
+}
+
+func (h *gatewayEventHandler) MapToIAMAuthPolicies() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		var requests []reconcile.Request
+		if gw, ok := obj.(*gateway_api.Gateway); ok {
+			policies := h.mapper.GatewayToIAMAuthPolicies(context.Background(), gw)
+			for _, p := range policies {
+				h.log.Infof("Gateway [%s/%s] resource change triggers IAMAuthPolicy [%s/%s] resource change", gw.Namespace, gw.Name, p.Namespace, p.Name)
+				requests = append(requests, reconcile.Request{NamespacedName: k8s.NamespacedName(p)})
+			}
+		}
+		return requests
+	})
+}
+
+func (h *gatewayEventHandler) MapToAccessLogPolicies() handler.EventHandler {
+	return handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		var requests []reconcile.Request
+		if gw, ok := obj.(*gateway_api.Gateway); ok {
+			policies := h.mapper.GatewayToAccessLogPolicies(context.Background(), gw)
+			for _, p := range policies {
+				h.log.Infof("Gateway [%s/%s] resource change triggers AccessLogPolicy [%s/%s] resource change", gw.Namespace, gw.Name, p.Namespace, p.Name)
```

Comment:
Why use square brackets around the Gateway and Policy namespaced names? We don't do this elsewhere in the codebase